From 22be4fbc56ce04a3652ea28b9f500a7a27d6ee85 Mon Sep 17 00:00:00 2001
From: Bruce Ashfield <bruce.ashfield@windriver.com>
Date: Wed, 14 Apr 2010 16:58:29 +0800
Subject: [PATCH] cleanup a.out header for export

The <linux/a.out.h> header contains "#ifdef linux".  GCC's includes
then creates its own copy of the installed header with this changed to
"#ifdef __linux__".  There should be no need for "#ifdef linux" in
Linux kernel headers, so this patch removes the conditional.  With
this patch, fixincludes (from GCC 4.1) makes no changes to the headers
installed by "make headers_install" on MIPS (I haven't tested on other
architectures).

Signed-off-by: Joseph Myers <joseph@codesourcery.com>
Acked-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 include/linux/a.out.h |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/include/linux/a.out.h b/include/linux/a.out.h
index e86dfca..47e647f 100644
--- a/include/linux/a.out.h
+++ b/include/linux/a.out.h
@@ -129,7 +129,6 @@ enum machine_type {
 #define SEGMENT_SIZE PAGE_SIZE
 #endif
 
-#ifdef linux
 #ifdef __KERNEL__
 #include <asm/page.h>
 #else
@@ -146,7 +145,6 @@ enum machine_type {
 #endif
 #endif
 #endif
-#endif
 
 #define _N_SEGMENT_ROUND(x) ALIGN(x, SEGMENT_SIZE)
 
-- 
1.6.5.2

